datasets/improved_shapes.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+"""NCS-Shapes.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1pNGCG013WE42axlPewwOPe-6v1p6ApAE
+"""
+
+from typing import Callable
+import torch
+
+def gaussian_mask(r, min, max, dist, device):
+    width = 2 * r + 1
+    g = (r - torch.arange(0, width, 1, device=device)) ** 2
+    grid = g + g.unsqueeze(0).T
+    img = torch.zeros(width, width, device=device)
+    img = torch.where((grid < max) & (grid > min), dist.sample((width, width)).to(device), img)
+    return img.bool()
+
+    
+def circle_new(size, p, device):
```

Comment:
I'm not a fan of this naming because "new" doesn't tell the user much (what does it do? why is it new?).
How about naming it something like "circle_double"? Feel free to call it something better! :)

Same comment applies to the names below.

Reply:
I will change the function names to circle_double or circle_improved or something along the lines from circle_new (and the corresponding names).

Reply:
Yes, the render_shape function in the improved_render is basically the corresponding render function for the new shapes function. I added them separately in case we wanted to use both.

Thanks!

Reply:
Right. Do they do different things? In that case, could you just give it an appropriate name and throw it into the `render` file? Then I agree it's a good idea to include both :)

Reply:
Right, I will do so